bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase1_x/TestMetrics.java

```diff
@@ -0,0 +1,243 @@
+  private BigtableConnection connection;
+
+  @Before
+  public void setUp() throws IOException {
+    Configuration configuration = new Configuration(false);
+    configuration.set(BigtableOptionsFactory.PROJECT_ID_KEY, TEST_PROJECT_ID);
+    configuration.set(BigtableOptionsFactory.INSTANCE_ID_KEY, TEST_INSTANCE_ID);
+    configuration.set(BigtableOptionsFactory.BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY, "true");
+    configuration.set(BigtableOptionsFactory.BIGTABLE_DATA_CHANNEL_COUNT_KEY, "1");
+    configuration.set(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY, "localhost:" + dataPort);
+    configuration.set(BigtableOptionsFactory.BIGTABLE_USE_GCJ_CLIENT, "true");
+    connection = new BigtableConnection(configuration);
+  }
+
+  @BeforeClass
+  public static void setUpServer() throws IOException {
+    try (ServerSocket s = new ServerSocket(0)) {
+      dataPort = s.getLocalPort();
+    }
+    server = ServerBuilder.forPort(dataPort).addService(fakeDataService).build();
+    server.start();
+  }
+
+  @AfterClass
+  public static void tearDownServer() throws InterruptedException {
+    if (server != null) {
+      server.shutdownNow();
+      server.awaitTermination();
+    }
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    connection.close();
+  }
+
+  @Test
+  public void readRows() throws IOException, InterruptedException {
+    FakeMetricRegistry fakeMetricRegistry = new FakeMetricRegistry();
+
+    BigtableClientMetrics.setMetricRegistry(fakeMetricRegistry);
+    long getTableStart = System.currentTimeMillis();
+    Table table = connection.getTable(TABLE_NAME);
+    long getTableTime = System.currentTimeMillis() - getTableStart;
+
+    long readRowsStart = System.currentTimeMillis();
+    Result result = table.get(new Get(new byte[2]));
+    long readRowsTime = System.currentTimeMillis() - readRowsStart;
+
+    ReadRowsRequest request = fakeDataService.popLastRequest();
+    assertEquals(FULL_TABLE_NAME, request.getTableName());
+
+    BigtableHBaseVeneerSettings bigtableSettings =
+        (BigtableHBaseVeneerSettings) connection.getBigtableSettings();
+    MetricsApiTracerAdapterFactory tracerFactory =
+        (MetricsApiTracerAdapterFactory)
+            bigtableSettings.getDataSettings().getStubSettings().getTracerFactory();
+
+    RpcMetrics readRow = tracerFactory.getMetrics().get("ReadRow");
+    Assert.assertNotNull(readRow);
+
+    AtomicLong readRowFailure =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.grpc.method.ReadRow.failure");
+    AtomicLong operationLatency =
+        fakeMetricRegistry.results.get(
+            "google-cloud-bigtable.grpc.method.ReadRow.operation.latency");
+    AtomicLong tableGetLatency =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.table.get.latency");
+
+    Assert.assertEquals(3, readRowFailure.get());
+    Assert.assertTrue(
+        operationLatency.get() <= readRowsTime + 50 || operationLatency.get() >= readRowsTime - 50);
```

Comment:
I dont think you need to extend the upper bound since it will always be bounded by your stopwatch

Reply:
Also it would be nice to split these assertions and add an assertion message so that when it fails we get more info than just expected true, but got false